R/scale-manual.r

```diff
@@ -124,6 +124,18 @@ manual_scale <- function(aesthetic, values = NULL, ...) {
     force(values)
   }
 
+  # order values according to breaks
+  args <- list(...)
+  if (is.vector(values) && is.null(names(values))
+      && "breaks" %in% names(args)) {
+    if (length(args[["breaks"]]) != length(values)) {
```

Comment:
I'm not particularly fond of this setup. It will force all dots to get evaluated which may have unintended consequences. Is there a reason why `breaks` was not added as a formal argument?

Reply:
I missed this. Good point. I agree.

@idno0001 You should add a `breaks` argument with default `waiver()` (just like here: https://github.com/tidyverse/ggplot2/blob/5e388e143f3d24bbeb6ce6b892cdb91deb5af6a2/R/scale-.r#L158-L159) and then explicitly hand this over to `discrete_scale()`. And the correct way to check whether breaks are set or not is then to test with `is.waive()`, I think. https://github.com/tidyverse/ggplot2/blob/40e8b6094e23cad19c8f06182af6730a963c205f/R/utilities.r#L180

Reply:
Thanks for the comments and apologies for the delayed response, I've made the suggested changes. Initially I didn't want to make such 'big' changes, but I see how it makes sense to add `breaks` as an argument.